### PR TITLE
CRUD operations to book controller

### PR DESCRIPTION
**Pull request by wleoguty91**
## Summary

- Added CRUD operations to book controller

## Trello Card

- https://trello.com/c/XAAyc91H/18-crear-las-operaciones-crud-de-book

### CODE REVIEW

File: src/main/java/wolox/training/controllers/BookController.java [hunk 1]
```diff
@@ -1,16 +1,58 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.server.ResponseStatusException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
 
 @Controller
```
Comment (leonplondon) [inline src/main/java/wolox/training/controllers/BookController.java]: Maybe in the future, when UI is no merged with business logic, you can use `@RestController` for a REST API

File: src/main/java/wolox/training/controllers/BookController.java [hunk 2]
```diff
@@ -1,16 +1,58 @@
...
 public class BookController {
 
+    @Autowired
+    private BookRepository bookRepository;
+
```
Comment (leonplondon) [inline src/main/java/wolox/training/controllers/BookController.java:25]: For best practices, we use to add a `@Autowired` constructor passing in required dependencies (constructor based injection), so if you notice your constructor is growing fast maybe is time to refactor o think about the truly goal of the controller.

**Comment by leonplondon** [comment]
Nice job!!!

File: src/main/java/wolox/training/controllers/BookController.java (hunk 2 above)
Comment (wleoguty91) [reply src/main/java/wolox/training/controllers/BookController.java:25]: I applied your suggestion, thank you :grinning:

File: src/main/java/wolox/training/controllers/BookController.java [hunk 3]
```diff
@@ -1,16 +1,61 @@
...
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
 
-@Controller
+@RestController
 public class BookController {
```
Comment (facucompe) [inline src/main/java/wolox/training/controllers/BookController.java:22]: Add `@RequestMapping("/books")` to the controller

**Review by facucompe — DISMISSED** [review]

File: src/main/java/wolox/training/controllers/BookController.java (hunk 3 above)
Comment (wleoguty91) [reply src/main/java/wolox/training/controllers/BookController.java:22]: Done!

File: src/main/java/wolox/training/controllers/BookController.java (hunk 2 above)
Comment (leonplondon) [reply src/main/java/wolox/training/controllers/BookController.java:25]: Remember to add `@Autowired` annotation  to constructor so other developers know this is a DI supplied argument.

Comment (wleoguty91) [reply src/main/java/wolox/training/controllers/BookController.java:25]: Done!

**Review by leonplondon — APPROVED** [review]
Nice job!

Keep going like that.

**Pull request merged**